docs/performance/productizing.rst

```diff
@@ -0,0 +1,116 @@
+.. _topics-to-watch-out-when-productizing-cratedb:
+
+###############################################
+ Topics to watch out when productizing CrateDB
+###############################################
```

Comment:
That's a bit of an outlier. Can you try to find a more concise headline and link label?

![image](https://github.com/user-attachments/assets/160b2c47-6986-4da4-90a5-ec49b24167c5) 
![image](https://github.com/user-attachments/assets/087cf9a0-6657-475f-935c-5fc4bb39e0f8)

Reply:
It can be made different on the navigation vs. on the page headline, when needed. In this case, I think both spots can benefit.

Reply:
Is "productizing" meant in the same sense like [Going into production](https://cratedb-guide--182.org.readthedocs.build/admin/going-into-production.html), but focusing on performance details instead of operational ones?

Reply:
What about a possible shorter "Optimize for production" in this case, or, alternatively, a more general "Production notes"?

Reply:
I changed it to the shorter "Key design considerations for scaling" now, but it seems it is still too long?

Reply:
It's still a bit long, yes. But we can also change it after an initial merge.